packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/NamedTypeSymbolProviders/ConstructorInitializerTests.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Tests.Common;
+using NUnit.Framework;
+
+namespace Microsoft.TypeSpec.Generator.Tests.Providers.NamedTypeSymbolProviders
+{
+    public class ConstructorInitializerTests
+    {
+        [Test]
+        public async Task CustomConstructorWithThisInitializerShouldHaveInitializerPopulated()
+        {
+            // Arrange
+            var compilation = await Helpers.GetCompilationFromDirectoryAsync();
+            await MockHelpers.LoadMockGeneratorAsync(compilation: async () => compilation);
+            var testClassSymbol = compilation.GetTypeByMetadataName("Sample.Models.TestClass");
```

Comment:
This should be removed, as well as the explicit construction of NamedTypeSymbolProvider. Let's move this into a new test class under Providers/ConstructorProviders. We should just construct the modelprovider and then get the constructor provider from the modelprovider.CustomCodeView.